src/libraries/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/ThrowHelpers.cs

```diff
@@ -0,0 +1,109 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Internal.Runtime.CompilerHelpers
+{
+    /// <summary>
+    /// These methods are used to throw exceptions from generated code.
+    /// For NativeAOT, the type and methods used need to be public
+    ///     as they constitute a public contract.
+    /// For CoreCLR, the type and methods are used as JIT helpers.
```

Comment:
```suggestion
```
This ifdef is not necessary (left over from ProjectN). The comment and the ifdef can be deleted, both here and in the native AOT copy.

Reply:
(The type can be internal in native AOT, just like it is in CoreCLR. We have number of similar internal types in native AOT.)